tests/functional/metrics/test_metrics.py

```diff
@@ -402,6 +406,66 @@ def test_conversion_metric(
         )
 
 
+class TestCumulativeMetric:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "purchasing.sql": purchasing_model_sql,
+            "metricflow_time_spine.sql": metricflow_time_spine_sql,
+            "semantic_models.yml": conversion_semantic_model_purchasing_yml,
+            "conversion_metric.yml": cumulative_metric_yml,
+        }
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"mock_purchase_data.csv": mock_purchase_data_csv}
+
+    def test_cumulative_metric(self, project):
+        # initial parse
+        runner = dbtRunner()
+        result = runner.invoke(["parse"])
+        assert result.success
+        assert isinstance(result.result, Manifest)
+
+        manifest = get_manifest(project.project_root)
+        metric_ids = set(manifest.metrics.keys())
+        expected_metric_ids_to_cumulative_type_params = {
+            "metric.test.weekly_visits": CumulativeTypeParams(
+                window=MetricTimeWindow(count=7, granularity=TimeGranularity.DAY),
+                period_agg=PeriodAggregation.AVERAGE,
```

Comment:
This could _probably_ be converted to unit test. I don't think it's necessary to do that as part of this PR though. I've been working on being able to unit test the reader clases in `schema_yaml_readers.py`, and it should be possible, but I think there is one kink to still work out first.